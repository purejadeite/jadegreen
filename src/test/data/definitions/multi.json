{
	"id": "book",
	"sheets": [
		{
			"comment": "2種類各1件①",
			"id": "main",
		    "name": "bean",
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "package", "row": 3, "column": 3},
		        {"id": "imports", "row": 4, "column": 3, "options": [{"class": "split", "splitter": "\n"}]},
		        {"id": "fields", "beginRow": 8, "endKeyId": "fieldName", "options": [{"class": "sort", "keyId": "fieldName"}],
		            "columns": [
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "method", "column": 3, "options": [{"class": "uppercamel"}]},
		                {"id": "static", "column": 3, "options": [{"class": "upperunderscore"}]},
		                {"id": "type", "column": 4},
		                {"id": "initial", "column": 5}
		            ]
		        }
		    ]
		},
		{
			"comment": "2種類各1件②",
			"id": "sub",
		    "name": "validation",
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "fields", "beginRow": 6, "endKeyId": "fieldName", "options": [{"class": "group", "keyId": "fieldName", "toId": "validators"}],
		            "columns": [
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "validationType", "column": 4, "options": [{"class": "mapping", "map": {"notnull": "NotNull"}, "lazy": true}]},
		                {"id": "arg0", "column": 5, "options": [{"class": "tonull"}]},
		                {"id": "arg1", "column": 6, "options": [{"class": "tonull"}]},
		                {"id": "arg2", "column": 7, "options": [{"class": "tonull"}]},
		                {"id": "arg3", "column": 8, "options": [{"class": "tonull"}]}
		            ]
		        }
		    ]
		}
	]
}