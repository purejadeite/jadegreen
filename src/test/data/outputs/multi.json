[{"package":"sdpj.scone.generated","className":"Sample","logicalClassName":"サンプル","imports":["java.io.Serializable","java.util.List","java.util.ArrayList","sdpj.scone.generated.LineBean"],"fields":[{"static":"FILE_EXIST_FLAG","initial":"Boolean.TRUE","logicalFieldName":"ファイル有無フラグ","method":"FileExistFlag","type":"Boolean","fieldName":"fileExistFlag"},{"static":"FILE_NAME","initial":"null","logicalFieldName":"ファイル名","method":"FileName","type":"String","fieldName":"fileName"},{"static":"FILE_NAME_JP","initial":"null","logicalFieldName":"ファイル名日本語","method":"FileNameJp","type":"String","fieldName":"fileNameJp"},{"static":"FILE_PATH","initial":null,"logicalFieldName":"ファイルパス","method":"FilePath","type":"String","fieldName":"filePath"},{"static":"LINES","initial":"new ArrayList<>()","logicalFieldName":"行情報","method":"Lines","type":"List<LineBean>","fieldName":"lines"}]},{"className":"Sample","logicalClassName":"サンプル(バリデーション)","fields":[{"validators":[{"arg3":null,"arg2":null,"arg1":null,"arg0":null,"logicalFieldName":"ファイル有無フラグ","validationType":"NotNull","fieldName":"fileExistFlag"}],"fieldName":"fileExistFlag"},{"validators":[{"arg3":null,"arg2":null,"arg1":null,"arg0":null,"logicalFieldName":"ファイルパス","validationType":"NotNull","fieldName":"filePath"},{"arg3":null,"arg2":null,"arg1":"10","arg0":"2","logicalFieldName":"ファイルパス","validationType":"length","fieldName":"filePath"}],"fieldName":"filePath"},{"validators":[{"arg3":null,"arg2":null,"arg1":null,"arg0":null,"logicalFieldName":"ファイル名","validationType":"NotNull","fieldName":"fileName"}],"fieldName":"fileName"},{"validators":[{"arg3":null,"arg2":null,"arg1":null,"arg0":null,"logicalFieldName":"行情報","validationType":"NotNull","fieldName":"lines"},{"arg3":null,"arg2":null,"arg1":null,"arg0":"100","logicalFieldName":"行情報","validationType":"maxlength","fieldName":"lines"},{"arg3":null,"arg2":null,"arg1":null,"arg0":null,"logicalFieldName":"行情報","validationType":"kana","fieldName":"lines"}],"fieldName":"lines"}]}]