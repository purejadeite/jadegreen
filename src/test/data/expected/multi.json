[
	{
		"logicalClassName": "複数定義・定義毎単一コンテンツ①",
		"className": "Sample",
		"package": "com.purejadeite.generated",
		"imports": [
			"JAVA.IO.SERIALIZABLE",
			"JAVA.UTIL.LIST",
			"JAVA.UTIL.ARRAYLIST",
			"COM.PUREJADEITE.GENERATED.LINEBEAN"
        ],
		"fields": [
			{
				"logicalFieldName": "ファイル有無フラグ",
				"fieldName": "fileExistFlag",
				"method": "FileExistFlag",
				"static": "FILE_EXIST_FLAG",
				"type": "Boolean",
				"initial": "Boolean.TRUE"
			},
			{
				"logicalFieldName": "ファイル名",
				"fieldName": "fileName",
				"method": "FileName",
				"static": "FILE_NAME",
				"type": "String",
				"initial": "null"
			},
			{
				"logicalFieldName": "ファイル名日本語",
				"fieldName": "fileNameJp",
				"method": "FileNameJp",
				"static": "FILE_NAME_JP",
				"type": "String",
				"initial": "null"
			},
			{
				"logicalFieldName": "ファイルパス",
				"fieldName": "filePath",
				"method": "FilePath",
				"static": "FILE_PATH",
				"type": "String",
				"initial": null
			},
			{
				"logicalFieldName": "行情報",
				"fieldName": "lines",
				"method": "Lines",
				"static": "LINES",
				"type": "List<LineBean>",
				"initial": "new ArrayList<>()"
			}
		]
	}
]